internal/controller/controllers/agent_controller.go

```diff
@@ -348,6 +363,35 @@ func (r *AgentReconciler) populateEventsURL(log logrus.FieldLogger, agent *aiv1b
 	return nil
 }
 
+func generateControllerLogsDownloadURL(baseURL string, clusterID string, authType auth.AuthType, filterBy ...string) (string, error) {
```

Comment:
I imagine some of the code here came from: https://github.com/openshift/assisted-service/blob/69bb1f6dae8c21aa31f00af04d3d2b1c0f1084ac/internal/controller/controllers/common.go#L266-L286


Do you think we can extract the common pieces into a function they both call?


Also, maybe it's worth moving this function into the `common.go` package where the `generateEventsURL` function is.

Reply:
That's a great idea! I discussed with Nick and we think the best approach might be to create a separate PR to refactor this function to contain the common pieces to limit the scope of this PR to just adding the cluster logs. What do you think?

Reply:
If it's a global function then it belongs to the same common file

Reply:
It's not being used as a global function yet, but if it will be later on then is it best to move it to the common file in this PR?